lib/rules/destructuring-assignment.js

```diff
@@ -10,6 +10,44 @@ const isAssignmentLHS = require('../util/ast').isAssignmentLHS;
 
 const DEFAULT_OPTION = 'always';
 
+function createSFCParams() {
+  const queue = [];
+
+  return {
+    push(params) {
+      queue.unshift(params);
+    },
+    pop() {
+      queue.shift();
+    },
+    get propsName() {
+      for (const params of queue) {
+        const props = params[0];
+        if (props && !props.destructuring && props.name) {
+          return props.name;
+        }
+      }
+      return null;
+    },
+    get contextName() {
+      for (const params of queue) {
+        const context = params[1];
+        if (context && !context.destructuring && context.name) {
+          return context.name;
+        }
+      }
+      return null;
+    }
```

Comment:
getters are slow and the indirection is less maintainable; let's just make these functions.
```suggestion
    propsName() {
      for (const params of queue) {
        const props = params[0];
        if (props && !props.destructuring && props.name) {
          return props.name;
        }
      }
      return null;
    },
    contextName() {
      for (const params of queue) {
        const context = params[1];
        if (context && !context.destructuring && context.name) {
          return context.name;
        }
      }
      return null;
    }
```

Reply:
ok, but why getters are slow and the indirection is less maintainable, please tell me more detail. please do tell me!

Reply:
They’re slow because it takes more effort for the engine to use them; they’re less maintainable because it looks like a property at the usage site, but it’s actually invoking a function. Making it a real function means you add `()` and suddenly everything is faster and more obvious. Getters and setters are bad and should be avoided.

Reply:
I haven't contributed code to this issue I originally reported (thanks @SyMind), but I am just reviewing what has been done so far and I feel compelled to say: that's a bad take. Getters and setters are quite a bit more performant than you might think (I use them in real-time applications even with emitters/streams to great effect), but I can respect that since you maintain this project you may have a different opinion.